server/game/abilities/keyword/ShieldedAbility.ts

```diff
@@ -0,0 +1,29 @@
+import AbilityHelper from '../../AbilityHelper';
+import TriggeredAbility from '../../core/ability/TriggeredAbility';
+import { Card } from '../../core/card/Card';
+import { KeywordName } from '../../core/Constants';
+import Game from '../../core/Game';
+import Contract from '../../core/utils/Contract';
+import { ITriggeredAbilityProps } from '../../Interfaces';
+
+export class ShieldedAbility extends TriggeredAbility {
+    public override readonly keyword: KeywordName | null = KeywordName.Shielded;
+
+    public static buildShieldedAbilityProperties(): ITriggeredAbilityProps {
+        return {
+            title: 'Shielded',
+            when: { onUnitEntersPlay: (event, context) => event.card === context.source },
+            immediateEffect: AbilityHelper.immediateEffects.giveShield()
+        };
+    }
+
+    public constructor(game: Game, card: Card) {
+        if (!Contract.assertTrue(card.isUnit())) {
```

Comment:
I think this check will break for leaders since they have `isUnit() === false` when not deployed. Just needs to be changed to `card.isUnit() || card.isLeader()` and should be fine